lib/validator.ex

```diff
@@ -27,11 +27,18 @@ defmodule Elixium.Validator do
     end
   end
 
-  @spec valid_index(number, number) :: :ok | {:error, {:invalid_index, number, number}}
-  defp valid_index(index, prev_index) when index > prev_index, do: :ok
+  #@spec valid_index(block, block) :: :ok | {:error, {:invalid_index, block, block}}
+  defp valid_index(block, last_block) do
+    if block.index > last_block.index do
+     :ok
+   else
+     Elixium.Store.Ledger.block_cache_validation(block)
```

Comment:
We should move this elsewhere, otherwise we could get flooded with blocks with wrong indexes that will fill up our cache by an attacker. A block should have at least a valid coinbase and a valid hash before being added to a cache